src/cli.rs

```diff
@@ -14,10 +12,47 @@ use crate::clonable_command::ClonableCommand;
 use crate::ignore::GlobMatcher;
 use crate::normal_path::NormalPath;
 
-/// A `ghci`-based file watcher and Haskell recompiler.
+/// Ghciwatch loads a GHCi session for a Haskell project and reloads it
+/// when source files change.
+///
+/// ## Examples
+///
+/// Load `cabal v2-repl` and watch for changes in `src`:
+///
+///     ghciwatch
+///
+/// Load a custom GHCi session and watch for changes in multiple locations:
+///
+///     ghciwatch --command "cabal v2-repl lib:test-dev" \
+///               --watch src --watch test
+///
+/// Run tests after reloads:
+///
+///     ghciwatch --test-ghci TestMain.testMain \
+///               --after-startup-ghci ':set args "--match=/OnlyRunSomeTests/"'
+///
+/// Use `hpack` to regenerate `.cabal` files:
+///
+///     ghciwatch --before-startup-shell hpack \
+///               --restart-glob '**/package.yaml'
+///
+/// Also reload the session when `.persistentmodels` change:
+///
+///     ghciwatch --watch config/modelsFiles \
+///               --reload-glob '**/*.persistentmodels'
+///
+/// Don't reload for `README.md` files:
+///
+///     ghciwatch --reload-glob '!src/**/README.md'
+#[allow(rustdoc::invalid_rust_codeblocks)]
 #[derive(Debug, Clone, Parser)]
-#[command(version, author, about)]
-#[command(max_term_width = 100)]
+#[command(
+    version,
+    author,
+    verbatim_doc_comment,
+    max_term_width = 100,
+    override_usage = "ghciwatch [--command SHELL_COMMAND] [--watch PATH] [OPTIONS ...]"
```

Comment:
`--command` and `--watch` are the really important options so I pulled them out; the default usage string was `ghciwatch [OPTIONS]` which isn't very useful.